src/Meziantou.Analyzer/Rules/UseShellExecuteAnalyzer.cs

```diff
@@ -0,0 +1,147 @@
+        messageFormat: "Use an overload of Process.Start that has a ProcessStartInfo parameter and set explicitly UseShellExecute value",
+        RuleCategories.Usage,
+        DiagnosticSeverity.Warning,
+        isEnabledByDefault: true,
+        description: "",
+        helpLinkUri: RuleIdentifiers.GetHelpUri(RuleIdentifiers.UseShellExecuteMustBeSet));
+
+    private static readonly DiagnosticDescriptor SetToFalseWhenRedirectingOutput = new(
+        RuleIdentifiers.UseShellExecuteMustBeSet,
+        title: "UseShellExecute must be explicitly set",
+        messageFormat: "Set UseShellExecute to false when redirecting standard input or output",
+        RuleCategories.Usage,
+        DiagnosticSeverity.Warning,
+        isEnabledByDefault: true,
+        description: "",
+        helpLinkUri: RuleIdentifiers.GetHelpUri(RuleIdentifiers.UseShellExecuteMustBeSet));
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+        ImmutableArray.Create(UseShellExecuteMustBeExplicitlySet);
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.EnableConcurrentExecution();
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+        context.RegisterCompilationStartAction(ctx =>
+        {
+            var analyzerContext = new AnalyzerContext(ctx.Compilation);
+            if (!analyzerContext.IsValid)
+                return;
+
+            ctx.RegisterOperationAction(analyzerContext.AnalyzeInvocation, OperationKind.Invocation);
+            ctx.RegisterOperationAction(analyzerContext.AnalyzeObjectCreation, OperationKind.ObjectCreation);
+        });
+
+    }
+
+    private sealed class AnalyzerContext(Compilation compilation)
+    {
+        private readonly INamedTypeSymbol? _processStartInfoSymbol = compilation.GetBestTypeByMetadataName("System.Diagnostics.ProcessStartInfo");
+
+        private readonly INamedTypeSymbol? _processSymbol = compilation.GetBestTypeByMetadataName("System.Diagnostics.Process");
+
+        public bool IsValid => _processStartInfoSymbol is not null;
+
+        public void AnalyzeInvocation(OperationAnalysisContext context)
+        {
+            var operation = (IInvocationOperation)context.Operation;
+            if (IsProcessStartInvocation(operation))
+            {
+                if (!operation.Arguments.Any(IsProcessStartInfo))
+                {
+                    // Calling Process.Start without ProcessStartInfo
+                    context.ReportDiagnostic(UseProcessStartOverload, operation);
+                }
+            }
+        }
+
+        public void AnalyzeObjectCreation(OperationAnalysisContext context)
+        {
+            var operation = (IObjectCreationOperation)context.Operation;
+            if (IsProcessStartInfoCreation(operation))
+            {
+                if (operation.Syntax is ObjectCreationExpressionSyntax { Initializer: {} initializer } )
```

Comment:
I don't think you need to use the syntax API. The syntax API is too brittle and harder to maintain with new versions of C#. Instead, you can use `operation.Initializer.Initializers`